egs/wsj/s5/steps/chaina/internal/get_train_schedule.py

```diff
@@ -0,0 +1,157 @@
+                        through the data.  The start/end x-values are at
+                        x=0/x=1, and other unspecified x-values are interpolated
+                        between known x-values.  You may specify different rules
+                        for different component-name patterns using
+                        'pattern1=func1 pattern2=func2', e.g. 'relu*=0,0.1,0
+                        lstm*=0,0.2,0'.  More general should precede less
+                        general patterns, as they are applied sequentially.""")
+
+    parser.add_argument("--num-scp-files", type=int, default=0, required=True,
+                        help="""The number of .scp files in the egs dir.""")
+    parser.add_argument("--schedule-out", type=str, required=True,
+                        help="""Output file containing the training schedule.  The output
+                        is lines, one per training iteration.
+                        Each line (one per iteration) is a list of ;-separated commands setting shell
+                        variables.  Currently the following variables are set:
+                        iter, num_jobs, inv_num_jobs, scp_indexes, frame_shifts, dropout_opt, lrate.
+                        """)
+
+    print(sys.argv, file=sys.stderr)
+    args = parser.parse_args()
+
+    return args
+
+def get_schedules(args):
+    num_scp_files_expanded = args.num_scp_files * args.frame_subsampling_factor
+    num_scp_files_to_process = int(args.num_epochs * num_scp_files_expanded)
+    num_scp_files_processed = 0
+    num_iters = ((num_scp_files_to_process * 2)
+                 // (args.num_jobs_initial + args.num_jobs_final))
+
+    with open(args.schedule_out, 'w', encoding='latin-1') as ostream:
+        for iter in range(num_iters):
+            current_num_jobs = int(0.5 + args.num_jobs_initial
+                                   + (args.num_jobs_final - args.num_jobs_initial)
+                                   * float(iter) / num_iters)
+            # as a special case, for iteration zero we use just one job
+            # regardless of the --num-jobs-initial and --num-jobs-final.  This
+            # is because the model averaging does not work reliably for a
+            # freshly initialized model.
+            if iter == 0:
+                current_num_jobs = 1
+
+            lrate = common_train_lib.get_learning_rate(iter, current_num_jobs,
+                                                       num_iters,
+                                                       num_scp_files_processed,
+                                                       num_scp_files_to_process,
+                                                       args.initial_effective_lrate,
+                                                       args.final_effective_lrate)
+
+            if args.dropout_schedule == "":
+                args.dropout_schedule = None
+            dropout_edit_option = common_train_lib.get_dropout_edit_option(
```

Comment:
common? 
should be libs.nnet3.train.dropout_schedule

Reply:
That's already imported in common_train_lib. See `steps/chain/train.py`.